src/Connectors/src/Connector.EFCore/MySqlDbContextOptionsExtensions.cs

```diff
@@ -216,13 +227,12 @@ private static MethodInfo FindUseSqlMethod(Type type, Type[] parameterTypes)
         var typeInfo = type.GetTypeInfo();
         var declaredMethods = typeInfo.DeclaredMethods;
 
-        foreach (var ci in declaredMethods)
+        foreach (var ci in declaredMethods.Where(method => method.Name.Equals("UseMySQL", StringComparison.InvariantCultureIgnoreCase)))
         {
             var parameters = ci.GetParameters();
-            if ((parameters.Length == 3 || parameters.Length == parameterTypes.Length) &&
-                ci.Name.Equals("UseMySQL", StringComparison.InvariantCultureIgnoreCase) &&
-                parameters[0].ParameterType.Equals(parameterTypes[0]) &&
-                parameters[1].ParameterType.Equals(parameterTypes[1]) &&
+            if (parameters.Length == parameterTypes.Length &&
+                parameters[0].ParameterType == parameterTypes[0] &&
+                parameters[1].ParameterType == parameterTypes[1] &&
```

Comment:
This should also check that the number of parameters is >= 2, to avoid a crash when this method is called with just one parameter.

Reply:
I hear what you're saying, but this should be caught at either dev-time or in the CI build. This method is private, only intended to work with 2 specific scenarios. Ideally we're making absolutely minimal changes in here before we get to the major refactor (which also needs to be part of 4.0)

Reply:
I agree with making minimal changes to existing code, because that's not the scope of this PR. However the diff shows you actually _removed_ the existing length check. The other 2 `FindUseSqlMethod` methods in the solution (untouched by this PR) contain that check as well, so for consistency I think we should keep it.

Reply:
Are you suggesting this:
`(parameters.Length == 3 || parameters.Length == parameterTypes.Length)` 
this:
`(parameters.Length >= 2 && parameters.Length == parameterTypes.Length)`
Or something different?

Reply:
Yes

Reply:
I realized the old code had `||` instead of `&&`, so it didn't validate properly either. This makes my argumentation invalid, so I'm okay with keeping it as-is.